awscli/customizations/cloudformation/yamlhelper.py

```diff
@@ -60,13 +62,31 @@ def _dict_representer(dumper, data):
     return dumper.represent_dict(data.items())
 
 
+def _add_yaml_1_1_boolean_resolvers(dumper_cls):
+    # CloudFormation treats unquoted values that are YAML 1.1 native
+    # booleans as booleans, rather than strings. In YAML 1.2, the only
+    # boolean values are "true" and "false" so values such as "yes" and "no"
+    # when loaded as strings are not quoted when dumped. This logic ensures
+    # that we dump these values with quotes so that CloudFormation treats
+    # these values as strings and not booleans.
+    boolean_regex = re.compile(
+        '^(?:y|Y|yes|Yes|YES|n|N|no|No|NO|'
+        'true|True|TRUE|false|False|FALSE'
+        '|on|On|ON|off|Off|OFF)$', re.X
+    )
+    boolean_first_chars = list(u'yYnNtTfFoO')
+    dumper_cls.add_implicit_resolver_base(
```

Comment:
What the difference between this resolver and default `implicit_resolvers` from `yaml/resolver.py`?

Reply:
There is no difference in the regex in that diff. However, I had to make a change to remove the single character yes's and no's because CloudFormation does not support that as a valid boolean YAML value.